scalr/data_source_scalr_module_versions.go

```diff
@@ -0,0 +1,124 @@
+package scalr
+
+import (
+	"context"
+	"errors"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/scalr/go-scalr"
+)
+
+func dataSourceModuleVersions() *schema.Resource {
+	return &schema.Resource{
+		Description: "Retrieves a list of module versions by module source or module id.",
+		ReadContext: dataSourceModuleVersionsRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description:  "The identifier of а module. Example: `mod-xxxx`",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				AtLeastOneOf: []string{"id", "source"},
+			},
+			"source": {
+				Description: "The source of a module.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"versions": {
+				Description: "The list of semantic versions.",
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Description: "The identifier of а module version. Example: `modver-xxxx`",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"version": {
+							Description: "The semantic version. Example: `1.2.3`.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"status": {
+							Description: "The status of a module version. Possible values: `ok`, `pending`, `not_uploaded`, `errored`",
```

Comment:
+ `pending_delete`

Reply:
API skips `pending_delete` in response.

Reply:
That's just a nitpick, but we do not know API internals here.
[API client has](https://github.com/Scalr/go-scalr/blob/50563b275996678d91337787cccc6776bdf42360/module_version.go#L47) `pending_delete` status, based on it's [source of truth](https://github.com/Scalr/fatmouse/blob/684a62e70ebd38598681f7b8432c8e5ca8cfdd61/taco/openapi/openapi-public.yml#L3017).

Reply:
I think this set of values are deprecated, because now pending_delete is invisible in api. It's like the webhook is deleted, but internally it's in deleting process.
Or we can show this value in ENUM, but clients never get this value.